ramp-engine/ramp_engine/base.py

```diff
@@ -67,6 +67,10 @@ def status(self):
     def status(self, status):
         self._status = status
 
+    def check_timeout(self):
```

Comment:
Should we make it an abstract method or could we implement something to be shared between the different worker type (the one in `aws/worker.py`)